protocol/messenger_messages.go

```diff
@@ -184,11 +199,16 @@ func (m *Messenger) DeleteMessageAndSend(ctx context.Context, messageID string)
 		return nil, err
 	}
 
-	message.Deleted = true
-	message.DeletedBy = deletedBy
-	err = m.persistence.SaveMessages([]*common.Message{message})
-	if err != nil {
-		return nil, err
+	response := &MessengerResponse{}
+	for _, messageToDelete := range messagesToDelete {
```

Comment:
As @borismelnik mentioned already, I think we should cover the new code with a bit of testing in this PR, or in a follow-up PR if this one needs to be merged asap.